msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs

```diff
@@ -119,9 +119,14 @@ void BuildNativeReferenceFlags (GccOptions gcc)
 					continue;
 				}
 
-				if (kind == NativeReferenceKind.Static) {
-					var libName = Path.GetFileName (item.ItemSpec);
+				var libName = Path.GetFileName (item.ItemSpec);
 
+				// existing (as of this commit) IDE versions will add a `.xcframework` directorty as a `Static` native reference
+				if ((kind == NativeReferenceKind.Static) && (Path.GetExtension (libName) == ".xcframework"))
```

Comment:
Thinking about dotnet support, I think it would be easier if the xcframework -> framework resolution was put into a new MSBuild target that ran before the _CompileToNative target, and rewrote the NativeReference item groups to be Frameworks pointing to the right path instead. This new target could then be re-used as-is in the dotnet code.

The end result is that there would be no changes to MTouchTaskBase (nor MmpTaskBase) classes, they would work as-is.

Reply:
What about dSYM files included in xcframeworks should we take this into account to put them in the output folder? here is an example of the structure of one xcframework

![unknown](https://user-images.githubusercontent.com/204671/98241887-95b20180-1f39-11eb-8542-a7d2032b4201.png)

Reply:
@dalexsoto bad example as it is not the _normal_ structure. That's how Xcode do them: https://github.com/spouliot/xcframework#tree

and PSPDFKit tweaks them https://pspdfkit.com/blog/2020/supporting-xcframeworks/#bcsymbolmaps-and-dsyms

The idea is to make them them (xcframework) transparent to the down-level tooling. It might be easy to deal with custom dSYM or not, but it's not a top priority.

Reply:
Interesting, I guess you need to enable it on demand since [Xcode 12.0 release notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-12-release-notes) state the following

> XCFrameworks can now include .dSYM and .bcsymbolmap debugging symbol files in your library bundle with the -debug-symbols flag. Run the command xcodebuild -create-xcframework -help for additional usage information. (64910707)

Reply:
interesting, I'll update my repo-sample to add the flag and see how it affects the files inside the xcframework

Reply:
added https://github.com/spouliot/xcframework/commit/9fe205228e52163df563d9d208a38f6bd3a2a8d1
it does make it clear / well-defined (inside the `.plist`) where the dSYM can be found